src/hotspot/share/cds/heapShared.cpp

```diff
@@ -443,8 +443,13 @@ bool HeapShared::is_lambda_proxy_klass(InstanceKlass* ik) {
   return ik->is_hidden() && (ik->name()->index_of_at(0, "$$Lambda+", 9) > 0);
 }
 
+bool HeapShared::is_string_concat_klass(InstanceKlass* ik) {
+  return ik->is_hidden() && ik->name()->starts_with("java/lang/String$$StringConcat");
```

Comment:
Can you add a comment in `StringConcatFactory`'s place where this name is defined, so others who change the generated class name will remember to update references here?

Reply:
Done.